### PR TITLE
Support for dynamic `kube_cluster` resources

### PR DESCRIPTION
**Pull request by tigrato**
This PR introduces Auth Server CRUD API for managing the lifecycle of dynamic `kube_cluster` resources.

Currently, `kube_cluster` resources cannot be manipulated without using the API, nor are they being listened to. It will be part of future developments.

# Future work
- The `kubernetes_service` will have to watch these `kube_cluster` resources.
- The discovery service will have to create these resources. 

Resource manipulation via `tctl` will be introduced later and is not part of the scope.

Part of #16135, #13376  
Related to  #12048

### CODE REVIEW

**Comment by r0mant** [comment]
@capnspacehook @timothyb89 Can you folks please review this?

File: lib/auth/clt.go [hunk 1]
```diff
@@ -860,7 +860,7 @@ func (c *Client) DeleteWebSession(ctx context.Context, user string, sid string)
 	return trace.Wrap(err)
 }
 
-// GenerateHostCert takes the public key in the Open SSH ``authorized_keys``
+// GenerateHostCert takes the public key in the Open SSH “authorized_keys“
```
Comment (timothyb89) [inline lib/auth/clt.go:863]: Still in nitpick mode, but kinda unusual to add the weird unicode quotes here (and 2x left double quote character at that). Maybe just regular double quotes?

```suggestion
// GenerateHostCert takes the public key in the Open SSH "authorized_keys"
```

File: lib/services/local/kube.go [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+	}
+	return nil
+}
+
+// DeleteKubernetesCluster removes the specified kubernetes cluster resource.
+func (s *KubernetesService) DeleteKubernetesCluster(ctx context.Context, name string) error {
+	err := s.Delete(ctx, backend.Key(kubernetesPrefix, name))
+	if err != nil {
+		if trace.IsNotFound(err) {
+			return trace.NotFound("kubernetes %q doesn't exist", name)
```
Comment (timothyb89) [inline lib/services/local/kube.go:124]: nit:
```suggestion
			return trace.NotFound("kubernetes cluster %q doesn't exist", name)
```

File: lib/services/local/kube.go [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+	}
+	return kubeClusters, nil
+}
+
+// GetKubernetesCluster returns the specified kubernetes cluster resource.
+func (s *KubernetesService) GetKubernetesCluster(ctx context.Context, name string) (types.KubeCluster, error) {
+	item, err := s.Get(ctx, backend.Key(kubernetesPrefix, name))
+	if err != nil {
+		if trace.IsNotFound(err) {
+			return nil, trace.NotFound("kubernetes %q doesn't exist", name)
```
Comment (timothyb89) [inline lib/services/local/kube.go:63]: nit:
```suggestion
			return nil, trace.NotFound("kubernetes cluster %q doesn't exist", name)
```

**Review by timothyb89 — APPROVED** [review]
Looks pretty solid, just a few nitpicks!

File: lib/services/role.go [hunk 4]
```diff
@@ -414,15 +416,16 @@ func applyValueTraitsSlice(inputs []string, traits map[string][]string, fieldNam
...
+//	env: ['{{external.groups}}']
 //
 // and groups: ['admins', 'devs']
 //
 // will be interpolated to:
 //
 // cluster_labels:
-//   env: ['admins', 'devs']
 //
+//	env: ['admins', 'devs']
```
[the remark this answers is not in the document]
Comment (tigrato) [reply lib/services/role.go:426]: You are right. I didn't detect these changes but it seems that go 1.19 introduced a lot of changes in gofmt and they result in weird comments formating

File: lib/auth/clt.go (hunk 1 above)
Comment (tigrato) [reply lib/auth/clt.go]: gofmt doing weird unicode replacements

File: lib/auth/grpcserver.go [hunk 5]
```diff
@@ -4369,6 +4369,96 @@ func (g *GRPCServer) AppendDiagnosticTrace(ctx context.Context, in *proto.Append
...
+func (g *GRPCServer) GetKubernetesClusters(ctx context.Context, _ *empty.Empty) (*types.KubernetesClusterV3List, error) {
+	auth, err := g.authenticate(ctx)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	clusters, err := auth.GetKubernetesClusters(ctx)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	var kubeClusters []*types.KubernetesClusterV3
```
Comment (capnspacehook) [inline lib/auth/grpcserver.go:4425]: Nit: could preallocate `kubeClusters` since you know how many elements you'll be putting in it

**Review by capnspacehook — APPROVED** [review]
Looks good, very straightforward

**Pull request merged**